### ISSUE TITLE
Hang when processing example in docstring

### ISSUE THREAD

**Issue by Andrew-Sheridan** [mitmproxy/pdoc#376]
#### Problem Description
pdoc hangs when processing certain examples in docstrings. This took me *ages* to pin down :)

#### Steps to reproduce the behavior:
Add this content to some file, foo.py:
```python

def foo():
    """Foo the foo.

    Examples:
        >>> class Foo(object):
        ...     pass
        >>> foo = Foo()
        >>> paths = foo.foo(dir_="src")
        >>> [str(p) for p in paths][:2]
        ['foo', 'bar']

    """
```

Run pdoc w/ google format
```shell
pdoc --docformat google --output-directory test-docs -- foo.py
```
- hangs :( 

Counter example:
Add this content to some file, bar.py:
```python

def bar():
    """Bar the bar.

    Examples:
        >>> class Bar(object):
        ...     pass
        >>> bar = Bar()
        >>> paths = bar.bar(dir_="src")

    """
```

```shell
pdoc --docformat google --output-directory test-docs -- foo.py
```
- works as expected :) 

#### System Information
Paste the output of "pdoc --version" here.
```console
❯ pdoc --version
pdoc: 11.0.0
Python: 3.7.9
Platform: Darwin-21.4.0-x86_64-i386-64bit
```

Thanks!

**Comment by Andrew-Sheridan** [comment]
Ok, so I tried this again in a fresh python3.9 virtual env
```shell
python3.9 -m venv venv
source venv/bin/activate
pip install --upgrade pip
pip install pdoc
```

and it doesnt hang when processing `foo.py`. It takes 45 seconds, but it completes

Processing `bar.py` takes ~1 second.

The examples are really fiction, the real code and docstrings are ~1000 lines.
Processing that never seems to finish - developers have been cancelling the execution after 10 minutes or so. I'll start one before dinner and report back.

**Comment by mhils** [comment]
Ugs - thank for the fantastic detective work, looks like we have a regex with exponential search behavior in markdown2. I'll take a look. :)

**Comment by mhils** [comment]
That indeed was the case! Fixed in 06beb3fcdac28d71ab058d57c995c69989e0489a, I'll file a PR upstream momentarily.
Fun fact: This bug [existed for 14 years](https://github.com/trentm/python-markdown2/blame/9c1897b2bce1058932b5d2bb9accd1a1d01b7508/lib/markdown2.py#L1060)! 😅

**Comment by andrewasheridan** [comment]
14 years! Amazing :D

Thank you for your efforts
